### PR TITLE
28 create member class closes #28

### LINKED ISSUE #28: Create Member Class

### PR DESCRIPTION
**Pull request by GoldHorizon**
Basic regular and executive member classes, both children of an abstract member class.  Not sure if it's exactly how a abstract class is supposed to be made so I just wrote the basic set and get methods for the first three attributes.

We will also need to track their rebate, membership renew date (?), list of items purchases, and possibly more.  We'll see, just check it out and see if this will work for now.

### CODE REVIEW

File: include/member.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    member();
+    member(std::string name, std::string member_type, int member_number);
+    virtual ~member();
+
+    // Setters
+    void name(std::string name);
+    void member_type(std::string member_type);
+    void member_number(int member_number);
+
+    // Getters
```
Comment (AnthonyKoutroulis) [inline include/member.h]: these should be constant, returning functions ie: 
 ```std::string name() const { return name_; } ```

File: include/member.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#ifndef MEMBER_H
+#define MEMBER_H
+
+class member
```
Comment (AnthonyKoutroulis) [inline include/member.h]: classes should use CapsFirstCamelCase

File: src/member.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    member_type = member_type_;
+    member_number = member_number_;
+}
+
+virtual ~member()
+{
+
+}
+
+// Getters
```
Comment (AnthonyKoutroulis) [inline src/member.cpp]: the _Getters_ should be constant, returning functions ie: 
 ```std::string name() const { return name_; } ```

**Comment by AnthonyKoutroulis** [comment]
Just minor name things, and I think your getters and setters got mixed up.  If you want you can name the setters ```void set_attribute(type) ``` and leave the getters as ```type attribute() const ```  if that seems more clear.

You can just commit and push the changes, no need to mess with the pull request.

**Comment by GoldHorizon** [comment]
Okay that all makes sense.  I forgot a few details like the `const` for the get methods, and wasn't sure about the style.

File: include/member.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    /// Default Destructor
+    virtual ~Member();
+
+    /// Setters
+    void set_name(std::string name);
+    void set_member_type(std::string member_type);
+    void set_member_number(int member_number);
+
+    /// Getters
```
Comment (AnthonyKoutroulis) [inline include/member.h]: These are still assigning.  They need to be returning functions ie:
``` std::string name() const { return name_; } ```

**Comment by AnthonyKoutroulis** [comment]
This needs to have ```testing``` as base, I'll close and recreate the request.

**Pull request closed without merging**